### ISSUE TITLE
ImportError: symbol not found in flat namespace (protobuf 3.20.2)

### ISSUE THREAD

**Issue by pietrodn** [protocolbuffers/protobuf#10571]
**What version of protobuf and what language are you using?**
Python, `protobuf==3.20.2`

**What operating system (Linux, Windows, ...) and version?**
macOS 12.6

**What runtime / compiler are you using (e.g., python version or gcc version)**
Python 3.10.6

**What did you do?**
```bash
python3 -m venv venv
. venv/bin/activate
pip install protobuf==3.20.2
python3 -c "import google.protobuf.descriptor"
```

**What did you expect to see**
Nothing (a successful import)

**What did you see instead?**
```
>>> import google.protobuf.descriptor
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".venv/lib/python3.10/site-packages/google/protobuf/descriptor.py", line 47, in <module>
    from google.protobuf.pyext import _message
ImportError: dlopen(.venv/lib/python3.10/site-packages/google/protobuf/pyext/_message.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace (__ZN6google8protobuf15FieldDescriptor12TypeOnceInitEPKS1_)
```

FYI a similar error is happening inside Google's `google-cloud-bigquery` library.

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**
1. The issue does not happen in a Linux environment (our CI).
2. The issue does not happen with `protobuf==3.20.1`.
3. The issue does not happen on Python 3.8.

**Comment by jcrm1** [comment]
Also getting this error (still on macOS 12.6, Python 3.10.6)

**Comment by madsjk816** [comment]
+1 here,  macOS 12.6, Python 3.10.6, protobuf==3.20.2 and Mac M1

**Comment by miohtama** [comment]
Same: macOS 12.6, Python 3.10.6, protobuf==3.20.2 and Mac M1.

Started to happen after updating Python version.

**Comment by miohtama** [comment]
Confirming downgrade works:

```
  • Updating protobuf (3.20.2 -> 3.20.1)
```

**Comment by cinnil** [comment]
Had this same issue pop up while installing lstein/stable-diffusion and downgrading solved it as well. Thanks!

> Confirming downgrade works:
> 
> ```
>   • Updating protobuf (3.20.2 -> 3.20.1)
> ```

**Comment by AlexSchultz-clumio** [comment]
3.19.5 also suffers from the same issue. Downgrading to 3.19.4 works.

**Comment by nikiz24** [comment]
in my case
`brew install protobuf`
`brew link --overwrite protobuf`
works. Others like downgrade pip protobuf does not work for me.

**Comment by nikiz24** [comment]
in my case
`brew install protobuf`
`brew link --overwrite protobuf`
works. Others like downgrade pip protobuf does not work for me.

**Comment by scomma** [comment]
> 3.19.5 also suffers from the same issue. Downgrading to 3.19.4 works.

Confirmed fix, thanks! Came here trying to install stable diffuser.

**Comment by bh2smith** [comment]
Issue also doesn't happen on python 3.9!

**Comment by Ported** [comment]
Running "pip install protobuf==3.19.4" to downgrade also worked for me.

**Comment by pro0517** [comment]
+1 here, macOS 11.2, Python 3.10.7, protobuf==3.20.2 and Mac M1

**Comment by lightnessofbein** [comment]
I used to have troubles with protobuf==3.20.2 (ImportError cause by some symbols) right after `import feast`
Downgrading to 3.20.1 solves the problem

**Comment by Roymprog** [comment]
I do experience the same problem with Python version 3.8.13 trying to using poetry. Perhaps it's related to [this issue](https://github.com/protocolbuffers/protobuf/issues/10618).

**Comment by pietrodn** [comment]
There's a [known security vulnerability](https://github.com/advisories/GHSA-8gq9-2x98-w8hf) that affects 3.20.1 and is fixed by 3.20.2. However, 3.20.2 is unusable because of this bug.

Google should deal with this with high priority.

**Comment by NicolasSompairac** [comment]
Simply came here to confirm that downgrading protobuf from 3.19.5 to 3.19.4 removed this error. I was getting this message when trying to use Stable Diffusion on my M2 Mac with Python 3.10.6.

**Comment by mayeut** [comment]
The issue is still there with 3.20.3
macOS 12.6, Python 3.10.7, protobuf==3.20.3 and Mac M1

**Comment by tommyjcarpenter** [comment]
For us, we didnt have to go to 3.19. but had to go protobuf (3.20.3 -> 3.20.1)

**Comment by deannagarcia** [comment]
Sorry about this issue. The problem is that the 3.19 and 3.20 branches only used the pure python implementation of protobuf. The newest update has added an incomplete C++ implementation for M1 binaries which has caused the crash.

To work around, you can set the environment variable PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python.

We are working on a fix now, but in order to get the most efficient python implementation we would recommend you upgrade to protobuf 4.21.7.

**Comment by deannagarcia** [comment]
I believe this issue should be fixed for 3.19.5, 3.19.6, 3.20.2, and 3.20.3. Please let me know if you are still experiencing issues.

The fix will force python 3.10 osx users to use pure python. If performance is important to you, we highly recommend you upgrade to 4.21.7.

**Comment by pietrodn** [comment]
I confirm the issue is fixed for 3.19.5, 3.19.6, 3.20.2, and 3.20.3 on Python 3.10.6, macOS 12.6, M1 Pro processor. 😄 

Thank you!

**Comment by nmariz** [comment]
> I confirm the issue is fixed for 3.19.5, 3.19.6, 3.20.2, and 3.20.3 on Python 3.10.6, macOS 12.6, M1 Pro processor. 😄

No it's not.

```python
$ python --version
Python 3.10.3

$ python -c "import google.protobuf; print(google.protobuf.__version__)"
3.20.3

$ python -c "import google.protobuf.descriptor"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/nmariz/tmp/venv/lib/python3.10/site-packages/google/protobuf/descriptor.py", line 47, in <module>
    from google.protobuf.pyext import _message
ImportError: dlopen(/Users/nmariz/tmp/venv/lib/python3.10/site-packages/google/protobuf/pyext/_message.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace (__ZN6google8protobuf15FieldDescriptor12TypeOnceInitEPKS1_)
```

And upgrading to 4.21.7 is not a workable solution, since the stubs are incompatible with 3.x. As well as the environment variable, specially for libraries that use Protobuf. I don't get it why the bug is closed and not fixed. Basically all macOS users will have to use a release with a known security issue (in the 3.20 series).

**Comment by pietrodn** [comment]
@nmariz, somehow the import works perfectly on my Mac with Python 3.10.6 and Protobuf 3.20.3. Very weird. 🤔

**Comment by deannagarcia** [comment]
I closed this bug because I

**Comment by deannagarcia** [comment]
Sorry you're still seeing this issue, I thought it was resolved.

I was able to reproduce with the steps in the original bug and after my change the error went away. When you use those steps are you still seeing the error?

Can you provide reproduction steps that include installing protobuf? Have you uninstalled/reinstalled?

**Comment by nmariz** [comment]
Thanks for considering my comment, but please ignore it.
Indeed doesn't raise the error now, because I was using a binary version with the C extension, that I've installed with pip. I think you've yanked that package from PyPI and it's not available anymore, so for macOS ARM64 only the pure Python is available.

**Comment by marinagalenko** [comment]
Hello!
Not sure if I should comment here since the issue is closed, but I'm experiencing the same issue.

```
sw_vers
ProductName:		macOS
ProductVersion:		13.0.1
BuildVersion:		22A400

python --version
Python 3.10.7

python -c "import google.protobuf; print(google.protobuf.__version__)"
3.20.3

python -c "import google.protobuf.descriptor"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/marina.galenko/Documents/GSL/hudl-videospa/tests/ui/venv/lib/python3.10/site-packages/google/protobuf/descriptor.py", line 47, in <module>
    from google.protobuf.pyext import _message
ImportError: dlopen(/Users/marina.galenko/Documents/GSL/hudl-videospa/tests/ui/venv/lib/python3.10/site-packages/google/protobuf/pyext/_message.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '__ZN6google8protobuf15FieldDescriptor12TypeOnceInitEPKS1_'

```

**Comment by fabiogra** [comment]
I solved downgrading Python to `3.9.16` (Apple M1 macOS 12.6)

```
Python 3.9.16 (main, Mar  8 2023, 04:29:24) 
[Clang 14.0.6 ] :: Anaconda, Inc. on darwin

>>> import google.protobuf; print(google.protobuf.__version__)
3.20.3
>>> import google.protobuf.descriptor
>>> 
```
